### PR TITLE
hw3

### PR DESCRIPTION
**Pull request by disaltzman**

### CODE REVIEW

**Comment by rhancockn** [comment]
For each question (Q#), write a short narrative answer. Your answers should be written in Markdown and saved in a `README.md` file in a `results` subdirectory of this repository. Clearly indicate question numbers by using headings.

File: results/README.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+The design with the highest efficiency has the presentation of the stimuli generally spaced apart well and with very rare overlap. There are also some large silent gaps to allow for the HRF to reset. 
+
+## Q2
+
+The best timing structure from the make_random_times.py script appears to have even higher variability than the RSFgen design (it is more efficient). I'm not sure what the units on the Y axis represent, but there are now events that are reaching higher values than there were in the RSFgen output. 
+
+
+## Q3
+
+Efficiency appears to have once again increased, even compared to when we controlled the ITI, as now we have more events at reaching 2 or 3 on the Y axis. For the RSFgen design, at most it reach 1, and the MRT version controlling for ITI had most events at 3 or 1.
```
Comment (rhancockn) [inline results/README.md]: Did you look at the efficiency values? Efficiency should decrease across these 3 cases.

File: results/README.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+## Q4
+
+The steps we took for Q2 and Q3 provide us with some ways to optimize our design, i.e. we can increase our ITI in order to allow the HRF to reset between stimulus presentations (incerease separability), and we can prevent presenting the same stimulus consecutively to maximize activation and reduce the chance of a habituated response (increase variance in predictor).
```
Comment (rhancockn) [inline results/README.md]: Habituation is an issue, but it's not one that affects design efficiency in these models which assume the responses are always linear and can be sustained for any length of time.

File: scripts/mrt_timing_01_block_100.sh [hunk 3]
```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+#!/usr/bin/env python
+N=1
+
+for i in `seq ${N}`; do \
+
+	make_random_timing.py -num_stim 3 -num_runs 1 \
```
Comment (rhancockn) [inline scripts/mrt_timing_01_block_100.sh]: It looks like you are generating an event related design here...

File: scripts/mrt_timing_01_block_100.sh [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+	-num_reps 20 \
+	-prefix stimt_${i} \
+	-seed $i \
+	-stim_dur 2 \
+	-min_rest 1 \
+	-max_rest 5 \
+	-max_consec 1
+
+
+	3dDeconvolve -nodata 300 1 -polort 1 \
```
Comment (rhancockn) [inline scripts/mrt_timing_01_block_100.sh]: ...and then using that in a block model. For the part on block designs, you need to generate the stimulus timing files differently. I had in mind just creating a text file by hand, e.g.
```
0 150
``` 

would give you the block onset times for condition A with 50 second block durations.

File: results/README.md [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+Efficiency has again decreased from the RSFgen design, even moreso than the previous step. I assume that is occuring because there is very little time for the HRF to reset between presentations of different stimulus types, which makes it more difficult to estimate its unique contribution.
+
+## Q4
+
+In order to increase efficiency, we can increase our ITI in order to allow the HRF to reset between stimulus presentations (incerease separability). The most optimal design generated by RSFgen had some lengthy intervals without stimulus presentation to allow for this. In addition, avoiding presenting the same stimulus multiple times in a row as it decreases variance.	  
+
+## Q5
+
+The average efficiency is greatest for the 20 second block (0.031), extremely similar for the 50 second block (0.03115), and worst for the 100 second block (.07035). However, in all cases the A-C	 contrast is more efficient than the A-B contrast. I assume it has something to do with how the frequency of the contrast for A-C is more ideal (around .01-.15 Hz) than the A-B contrast, which is getting filtered out by the HRF more.
```
Comment (rhancockn) [inline results/README.md]: A-C should have worse efficiency, particularly for 100s. Did you switch your results?

File: results/README.md [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+
+## Q6
+
+Increasing the polort value decreased the efficiency of the design. I assume that the polort function is filtering the data somehow, and by increasing it, we are removing too much of the actual signal instead of just removing noise. If it is removing noise created by drift over the course of the block, it should affect it more seriously with a longer block (more time for drift to accrue) than a shorter block like 20 seconds.
```
Comment (rhancockn) [inline results/README.md]: Very good.

File: results/README.md (hunk 5 above)
Comment (disaltzman) [reply results/README.md]: Yep, just checked and I made a typo here. A-B was always more efficient.

**Pull request closed without merging**